### ISSUE TITLE
no-unused-vars is complaining when not true

### ISSUE THREAD

**Issue by MikaAK** [eslint/eslint#3917]
I was writing out a test spec and I noticed that my eslint went off so I thought I would report a bug.

__*ESLint Version `1.5.1`*__

### Error
```bash
2:7  error  "$httpBackend" is defined but never used  no-unused-vars
```

### Reproduction Code
```javascript
describe('CustomizeTable Directive', function() {
  var $httpBackend

  beforeEach(module('edvisor'))
  beforeEach(inject(function(_$httpBackend_) {
    $httpBackend = _$httpBackend_
  }))

  it('takes in a modelTypeId and fetches tablesPrefs')

  it('sorts fields from fetch into visible and nonvisible')

  it('displays sorted fields in two columns')

  it('allows checking and unchecking of fields without moving')

  it('resets on cancel')

  it('saves and updates on submit')
})
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by michaelficarra** [comment]
It's not read. The write is not observable. This is an unused variable. Read the documentation for more info: http://eslint.org/docs/rules/no-unused-vars

**Comment by MikaAK** [comment]
Ah ok! Thanks
